streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -478,9 +479,11 @@ void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
 
         commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
 
+        // We do not need to enforce checkpointing upon suspending a task: if it is resumed later we just
+        // proceed normally; if it is closed we would checkpoint then
         for (final Task task : revokedTasks) {
             try {
-                task.postCommit();
+                task.postCommit(false);
```

Comment:
Inspired by https://github.com/apache/kafka/pull/8962/files#diff-f0037ae2fd44bdd0d84f3ef37b43bc05R277, I think we should actually enforce checkpoint (i.e. set to `true`) when suspending since the suspended task may not be closed but recycled. cc @ableegoldman 

I will push a new commit changing to true.